### PR TITLE
New Rule: Attachment html base64 script source

### PR DESCRIPTION
**Pull request by morriscode**
82 messages found 
9 Already Flagged

### CODE REVIEW

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+name: "Attachment: HTML Smuggling with base64 script source"
+description: |
+  Potential HTML smuggling attacks from new senders.
+  This rule detects HTML files as attachments or within archives that contain base64 script sources from new or outlier senders. 
+references:
+
```
Comment (aidenmitchell) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:6]:
```suggestion
```

**Comment by morriscode** [comment]
Pending sender_profile release

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          or .file_type == "html"
+          or .content_type == "text/html"
+        )
+        and any(file.explode(.),
+                length(.scan.strings.strings) < 20
+                and any(.scan.strings.strings,
+                        regex.icontains(., 'script src="data:text/html;base64')
+                )
+        )
+    )
```
Comment (rw-access) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:22]: We also have `file.parse_html` where you can search the raw text. Doesn't help for the archive case unfortunately. Once that's out, we can use that here to make it simpler.

Comment (morriscode) [reply detection-rules/attachment_html_smuggling_with_base64_script_source.yml:22]: Sweeet thanks for the reminder, I think this came out before that was ready. I'll check it out!

**Comment by morriscode** [comment]
Test volume and resulting actions look solid, this is safe to ship.

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    any(attachments,
+        (
+          .file_extension in~ ("html", "htm", "shtml", "dhtml")
+          or .file_type == "html"
+          or .content_type == "text/html"
+        )
+        and any(file.explode(.),
+                length(.scan.strings.strings) < 20
+                and any(.scan.strings.strings,
+                        regex.icontains(., 'script src="data:text/html;base64')
```
Comment (aidenmitchell) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:19]:
```suggestion
                        regex.icontains(., 'src="data:text/html;base64')
```

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+           (.file_extension in~ $file_extensions_common_archives)
+           and any(file.explode(.),
+                   (
+                     .file_extension in~ ("html", "htm", "shtml", "dhtml")
+                     or ..file_type == "html"
+                     or ..content_type == "text/html"
+                   )
+                   and length(.scan.strings.strings) < 20
+                   and any(.scan.strings.strings,
+                           regex.icontains(., 'script src="data:text/html;base64')
```
Comment (aidenmitchell) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:33]:
```suggestion
                           regex.icontains(., 'src="data:text/html;base64')
```

**Review by aidenmitchell — CHANGES_REQUESTED** [review]

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
+name: "Attachment: HTML Smuggling with base64 script source"
```
Comment (aidenmitchell) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:1]:
```suggestion
name: "Attachment: HTML Smuggling with base64 source"
```

File: detection-rules/attachment_html_smuggling_with_base64_script_source.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+description: |
+  Potential HTML smuggling attacks from new senders.
+  This rule detects HTML files as attachments or within archives that contain base64 script sources from new or outlier senders.
```
Comment (aidenmitchell) [inline detection-rules/attachment_html_smuggling_with_base64_script_source.yml:4]:
```suggestion
  This rule detects HTML files as attachments or within archives that contain base64 sources from new or outlier senders. 
```

**Comment by morriscode** [comment]
Adding unsolicited and high trust sender negations.

**Pull request closed without merging**